juju/model.py

```diff
@@ -0,0 +1,167 @@
+import json
+from collections.abc import Mapping
+from functools import lru_cache
+from subprocess import run, PIPE
+
+
+class Model:
+    def __init__(self, relation_names, backend):
+        self.relations = RelationMap(relation_names, backend)
+        self._backend = backend
+
+    @property
+    def app(self):
+        return self._backend.local_app
+
+    @property
+    def unit(self):
+        return self._backend.local_unit
+
+    def relation(self, relation_name):
+        """Return the first Relation object for the named relation, or None.
+
+        This is a convenience method for the case where only a single related
+        app is supported. If there are more than one remote apps on the relation,
+        it will raise a TooManyRelatedApps model error. If no there are no remote
+        apps on the relation, it will return None. Otherwise, it is equivalent
```

Comment:
"no there are no".. but this is also reading a bit confusingly for a reason I can quite put my finger on. Maybe we can simplify a bit:
```
This convenience method returns None if the relation is not established, or
the related app if the relation is established only once. If this same relation
is established multiple times the error TooManyRelatedApps is raised.
```
How does that sound?